AEPServices/Sources/ui/fullscreen/FullScreenUIHandler.swift

```diff
@@ -0,0 +1,205 @@
+                         */
+                        do {
+                            try FileManager.default.createDirectory(atPath: cacheFolderURL!.path, withIntermediateDirectories: true, attributes: nil)
+                            try self.payload.write(toFile: tempHTMLFile!.path, atomically: true, encoding: .utf8)
+                            useTempHTML = true
+                        } catch {
+                            Log.debug(label: self.LOG_PREFIX, "Failed to save the temporary HTML file for fullscreen message \(error)")
+                            return
+                        }
+                    }
+                }
+                // load the HTML string on WKWebview. If we are using the cached images, then use
+                // loadFileURL:allowingReadAccessToURL: to load the html from local file, which will give us the correct
+                // permission to read cached files
+                if useTempHTML {
+                    wkWebView.loadFileURL(URL.init(fileURLWithPath: tempHTMLFile?.path ?? ""), allowingReadAccessTo: URL.init(fileURLWithPath: cacheFolder?.path ?? ""))
+                } else {
+                    wkWebView.loadHTMLString(self.payload, baseURL: Bundle.main.bundleURL)
+                }
+
+                let keyWindow = self.getKeyWindow()
+                keyWindow?.addSubview(wkWebView)
+                UIView.animate(withDuration: 0.3, animations: {
+                    var webViewFrame = wkWebView.frame
+                    webViewFrame.origin.y = 0
+                    wkWebView.frame = webViewFrame
+                }, completion: nil)
+            }
+        }
+        self.listener?.onShow(message: self.message)
+    }
+
+    /// Dismiss the message, and call the OnDismiss() on the listener.
+    func dismiss() {
+        DispatchQueue.main.async {
+            self.monitor.dismissed()
+            self.dismissWithAnimation(animate: true)
+            self.listener?.onDismiss(message: self.message)
+
+            // remove the temporary html if it exists
+            guard var cacheFolder: URL = self.fileManager.getCacheDirectoryPath() else {
+                return
+            }
+            cacheFolder.appendPathComponent(self.DOWNLOAD_CACHE)
+            cacheFolder.appendPathComponent(self.TEMP_FILE_NAME)
+            cacheFolder.appendPathComponent(self.HTML_EXTENSION)
+            let tempHTMLFilePath = cacheFolder.absoluteString
+
+            do {
+                try FileManager.default.removeItem(atPath: tempHTMLFilePath)
+            } catch {
+                Log.debug(label: self.LOG_PREFIX, "Unable to dismiss \(error)")
+            }
+        }
+    }
+
+    /// Opens the resource at the URL specified.
+    func openUrl(url: String) {
+        if !url.isEmpty {
```

Comment:
No need for the empty check if you will be guarding on the URL initializer because the initializer returns nil if the string is empty. So the guard suffices. Also, we do not use `URL.init(...)`. Simply use the `URL(string: ...)`

`guard let url = URL(string: url) else { return }`